File: frontend/src/types/models/ChannelModel.ts
enum ChannelStatus {
  Normal,
  Limited,
  Blocked,
}

interface ChannelModel {
  id: string;
  name: string;
  description?: string;
  created?: string;

  status: ChannelStatus;

  subscribes: number;

  isSubscripted: boolean;

  iconUrl: string;
  bannerUrl?: string;
}

export default ChannelModel;

export { ChannelStatus };

File: backend/BYTUBE/Models/ChannelModels/ChannelModel.cs
﻿using static BYTUBE.Entity.Models.Channel;

namespace BYTUBE.Models.ChannelModels
{
    public class ChannelModel
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public int Subscribes { get; set; }

        public ActiveStatus Status { get; set; }

        public bool IsSubscripted { get; set; }

        public string IconUrl { get; set; }
    }
}
